### PR TITLE
Update allow_redirects docstrings defaults

### PR DESCRIPTION
**Pull request by geokala**

### CODE REVIEW

File: requests/api.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def request(method, url, **kwargs):
         before giving up, as a float, or a :ref:`(connect timeout, read
         timeout) <timeouts>` tuple.
     :type timeout: float or tuple
-    :param allow_redirects: (optional) Boolean. Enable/disable GET/OPTIONS/POST/PUT/PATCH/DELETE/HEAD redirection. Defaults to ``True``.
+    :param allow_redirects: (optional) Boolean. Enable/disable GET/OPTIONS/POST/PUT/PATCH/DELETE/HEAD redirection. Defaults to ``False`` for HEAD, ``True`` for all others.
     :type allow_redirects: bool
```
Comment (Lukasa) [inline requests/api.py:37]: This is not correct. `requests.head` defaults this to `False`, but all other uses (including `requests.request('HEAD', ...)` default it to `True`.

File: requests/sessions.py [hunk 2]
```diff
@@ -465,7 +465,8 @@ def request(self, method, url,
             data before giving up, as a float, or a :ref:`(connect timeout,
             read timeout) <timeouts>` tuple.
         :type timeout: float or tuple
-        :param allow_redirects: (optional) Set to True by default.
+        :param allow_redirects: (optional) Set to False by default for HEAD,
+            True by default for all others.
```
Comment (Lukasa) [inline requests/sessions.py:469]: Same note here, this docstring does not require updating.

File: requests/api.py (hunk 1 above)
Comment (geokala) [reply requests/api.py:37]: OK, understood- and sorry for the delay, busy week. I'll look to get this updated before the end of the weekend.

Comment (Lukasa) [reply requests/api.py:37]: No worries: we're all busy people!

**Comment by Unit03** [comment]
This is still relevant imho, do you think we can reopen this PR, @kenneth-reitz @Lukasa ?

**Comment by geokala** [comment]
Well, that's a blast from the past. I'm happy to fix it- does it just require the removal of that second, incorrectly updated, docstring?

**Comment by Unit03** [comment]
@geokala I suppose it is. :) If it's of no relevance for you at this point, I will be equally happy to take this over.

**Comment by henribru** [comment]
Just ran into this myself, would be great if this could be merged so as to not cause confusion.

**Comment by geokala** [comment]
@Unit03 Happy for you to take it over if you're happy to.

**Comment by Unit03** [comment]
@geokala I will do just that, then, thank you for your input. :)

@Lukasa I've addressed your remarks and opened a new PR (to not take more of @geokala's time): #5127 - but I'd be ok to contribute this change in some other way if you think it's appropriate.

**Pull request closed without merging**